(DOCSP-27741) Bearer Auth Example for Custom & Generated Endpoints (#336)
mongodb/docs-app-services: source/data-api/custom-endpoints.txt

like an API key or JWT. You can also configure other custom
authentication schemes to fit your application's needs.

For examples of how to authenticate requests, see :ref:`Authenticate
Data API Requests <data-api-authenticate-requests>`.

.. tabs::

   .. tab:: Application
      :tabid: auth-credentials

      Application authentication requires users to log in with an
      authentication provider that you have enabled for your App.
      Requests can either include an access token granted by the

>       authentication provider or the credentials the user would log in

Author: Hmm I think one could always assume in our docs that "Auth Provider" refers to the things built-in to App Services vs the system that handles the authentication. Can definitely clarify here though!

>       with (e.g. their email and password).

Reviewer: might make sense to include the fact that this can be API key. i imagine passing API key in the auth is more common than username and password (for which i can think of no realistic use case)

Authenticate the Request
~~~~~~~~~~~~~~~~~~~~~~~~

If an endpoint is configured to use Application Authentication then you
must include a valid user access token or login credentials with every
request.

> In general, prefer to use an access token over login credentials when
> possible. The token lets you run multiple requests without

Reviewer: ```suggestion
In general, use an access token over login credentials when
possible. The token lets you run multiple requests without
```

re-authenticating the user. It also lets you send requests from a web

> browser or other environment that enforces :wikipedia:`CORS

Reviewer: ```suggestion
browser that enforces :wikipedia:`CORS
```

i think we can remove mention of other envs b/c per the linked wikipedia article, CORS is specifically for web browsers

<Cross-origin_resource_sharing>`.

> To use an access token, authenticate the user through an authentication
> provider and then include the access token in the request's

Reviewer: i rec adjusting phrasing to something like this to make clear the role of app services as an intermediary here.

```suggestion
To use an access token, authenticate the user into an App Services App through an authentication provider.
Then include the access token returned by App Services Authentication in the request's
```

Reviewer: (prob should adjust the janky line breaks if going w this phrasing)

Authorization header using a Bearer token scheme. For more information
on how to acquire and use an access token, see :ref:`Bearer Token
Authentication <data-api-bearer-authentication>`.

.. code-block:: shell
   :emphasize-lines: 2

   curl -X GET \
      -H 'Authorization: Bearer <AccessToken>' \
      -H 'Content-Type: application/json' \
      https://data.mongodb-api.com/app/myapp-abcde/endpoint/hello

> Alternatively, you can include valid login credentials for the user in

Reviewer: ```suggestion
Alternatively, you can include login credentials for the user in
```

nit: don't think nec
